service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/RepresentativesInbox.java

```diff
@@ -47,11 +56,44 @@
                 .collect(Collectors.toCollection(LinkedHashSet::new))
         );
 
-
         return emails;
     }
 
-    private boolean shouldSend(RepresentativeServingPreferences preference, Element<? extends WithSolicitor> element) {
+    public Set<?> getNonSelectedRespondentsRecipients(
+        RepresentativeServingPreferences servingPreferences,
+        CaseData caseData,
+        List<Element<Respondent>> respondentsSelected,
+        Function<Element<Representative>, ?> mapperFunction) {
+
+        Set<UUID> representativeIds = caseData.getAllRespondents().stream()
+            .flatMap(respondentElement -> respondentElement.getValue().getRepresentedBy()
+                .stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        Set<UUID> selectedRepresentativeIds = respondentsSelected.stream()
+            .flatMap(otherElement -> otherElement.getValue().getRepresentedBy().stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        return caseData.getRepresentativesElementsByServedPreference(servingPreferences)
+            .stream()
+            .filter(representativeElement -> representativeIds.contains(representativeElement.getId()))
+            .filter(representativeElement -> !selectedRepresentativeIds.contains(representativeElement.getId()))
+            .map(mapperFunction)
```

Comment:
Do you really need this mapperFunction?
If so, looks like it should be a constant in this class rather than something a caller passes as a parameter.

Reply:
It has to be passed as we are using it for different serving preferences. so we need address for POST preference and email for others